tests/foreman/ui/test_host.py

```diff
@@ -814,6 +814,166 @@ def test_negative_remove_parameter_non_admin_user(self):
+            content_view = content_view.read()
+            promote(content_view.version[0], filter_lc_env.id)
+        # Create two host groups
+        hg = entities.HostGroup(organization=[org]).create()
+        filter_hg = entities.HostGroup(organization=[org]).create()
+
+        # Create new role
+        role = entities.Role().create()
+
+        # Create lifecycle environment permissions and select one specific
+        # environment user will have access to
+        env_permissions_entities = entities.Permission(
+            resource_type='Katello::KTEnvironment').search()
+        user_env_permissions = [
+            'promote_or_remove_content_views_to_environments',
+            'view_lifecycle_environments'
+        ]
+        user_env_permissions_entities = [
+            entity
+            for entity in env_permissions_entities
+            if entity.name in user_env_permissions
+        ]
+        entities.Filter(
+            organization=[org],
+            permission=user_env_permissions_entities,
+            role=role,
+            # allow access only to the mentioned here environment
+            search='name = {0}'.format(filter_lc_env.name)
+        ).create()
+
+        # Add necessary permissions for content view as we did for lce
+        cv_permissions_entities = entities.Permission(
+            resource_type='Katello::ContentView').search()
+        user_cv_permissions = [
+            'promote_or_remove_content_views',
+            'view_content_views',
+            'publish_content_views',
+        ]
+        user_cv_permissions_entities = [
+            entity
+            for entity in cv_permissions_entities
+            if entity.name in user_cv_permissions
+        ]
+        entities.Filter(
+            organization=[org],
+            permission=user_cv_permissions_entities,
+            role=role,
+            # allow access only to the mentioned here cv
+            search='name = {0}'.format(filter_cv.name)
+        ).create()
+
+        # Add necessary permissions for hosts as we did for lce
+        host_permissions_entities = entities.Permission(
+            resource_type='Host').search()
+        user_host_permissions = [
+            'create_hosts',
+            'view_hosts',
+        ]
+        user_host_permissions_entities = [
+            entity
+            for entity in host_permissions_entities
+            if entity.name in user_host_permissions
+        ]
+        entities.Filter(
+            organization=[org],
+            permission=user_host_permissions_entities,
+            role=role,
+            search='hostgroup_fullname = {0}'.format(filter_hg.name)
+        ).create()
+
+        # Add necessary permissions for host groups as we did for lce
+        entities.Filter(
+            organization=[org],
+            permission=entities.Permission(name='view_hostgroups').search(),
+            role=role,
+            search='name = {0}'.format(filter_hg.name)
+        ).create()
```

Comment:
That's a lot of filters... How about creating them in a loop? That'd save us 50-100 lines at least and made a test a bit more readable. You could create a list similar to your `host_entities`, with keys like 'resource_type', 'names', 'search_query' etc?

Reply:
I thought about this, but it is not so easy as we have different approach per different filters as for a few of them we need perm names, for some only types and also we have different entities search criterias